### ISSUE TITLE
Second hand

### ISSUE THREAD

**Issue by android272** [teeworlds/teeworlds#1401]
If you look at most fan art of Tees and even art that is posted of tees on the main website they have two arms.  I suggest taking current arm from the skin and adding it to the top of the body as well as the back.  

When a Tee is moving right the their hand and the gun would be on top, I think it looks weird when a character in a side scroller just flips the sprite and the gun they were just holding in there "left hand" is now in their "right", and when the Tee is moving left the gun would be in the back and their "other hand" is on the top.

When the Tee is holding the pistol the left hand is free. I believe it would look very nice if the Tee's arm swung back and forth as it walked. 

![tee3](https://cloud.githubusercontent.com/assets/5912527/9835200/1059cf66-59ac-11e5-8c14-a8d4c1cac4bf.png)

Bellow is what each gun would look like with this system.
When looking right from top to bottom the Tee would stack: Right Leg, Right Hand, Gun, Eyes, Body, Left Hand, Left Leg

When looking left from top to bottom the Tee would stack: Left Leg, Left Hand, Eyes, Body, Gun, Right Hand, Right Leg

With two handed weapons the Tee's left hand would take the position of the current hand out in front of the Tee and the right hand would have to placed back at the trigger.

Note: I believe that the guns in the pictures below are much bigger then they would be in the game. The beam rifle would not cover that much of the Tee's face.  If this is still a problem then perhaps the Tee's eyes could be moved up a little for these guns.  Please don't through out this idea because the tees face is slightly covered up with bigger guns.

![pistol](https://cloud.githubusercontent.com/assets/5912527/9835186/af48dd48-59ab-11e5-8c48-4c9b5af639eb.png)
![shotgun](https://cloud.githubusercontent.com/assets/5912527/9835187/b4f83cde-59ab-11e5-9c75-3377ef6c52f2.png)
![thumper](https://cloud.githubusercontent.com/assets/5912527/9835188/badb85e8-59ab-11e5-8141-73c59c2c35a5.png)
![beam rifle](https://cloud.githubusercontent.com/assets/5912527/9835190/bee439f0-59ab-11e5-8c42-937a99505715.png)

The hook should be held in the left hand while the Tee holds their gun in their right hand.  It has been brought up that this will look impossible for the Tee to hold for example the rocket luncher or the shot gun with only one hand.  My response to this is that Teeworlds is a more cartoon world and it is not that big of a stretch of the imagination to believe that Tees can hold a rocket luncher in one hand. Even though this would totally rip our arms off if we were to try to attempt this.

![tee5](https://cloud.githubusercontent.com/assets/5912527/9835191/c439c802-59ab-11e5-9cbc-25e20efc9892.png)

The flag as well should be placed in the Tees off hand.

![flag](https://cloud.githubusercontent.com/assets/5912527/9835240/8e651e7c-59ae-11e5-97d6-67032554bdcf.gif)

**Comment by TheRealOrangus** [comment]
I didn't get.. You suggest to change the game only graphically, or the gameplay too?
Also, when a tee would rotate fast while holding some gun, it would look just crap, isn't hard to imagine that.

**Comment by android272** [comment]
This suggestion is only graphically. I don't see how this would affect gameplay other than I think the Tees would look better.

"Also, when a tee would rotate fast while holding some gun, it would look just crap, isn't hard to imagine that."

This is correct when the player has the the sensitivity way up high it would look like crap... but then again there are high-sens players today and in my opinion they look like crap now.  So either everyone looks like crap as the game is now, or make the game look better and high-sens players still look like crap. Which do you prefer?

**Comment by TheRealOrangus** [comment]
Yes, but I said when they rotate fastly, which happens for every player in teeworlds, by _"fast"_ I did not mean **super-high-crazy megaspin**, but the fastest rotation speed of a player.

> ... in my opinion they look like crap now.

Still, it is less of crapness now for a super-high-crazy megaspin player, than for a regular newb with this flip stuff.

Currently, movement, rotation, animations and such stuff, are very smooth now, with that flipping, it will be not smooth anymore.

**Comment by android272** [comment]
If you did not notice this thread is about adding a second hand to the tee.  If you are holding a pistol and you look left then tee tee would have a free hand swinging back and forth.  If you look left then the gun would be over the tees body and you would not see the second hand.  If you are holding a two handed weapon and you are looking left you would see one hand out holding the gun for suport above the tees body. If you are looking right with a two handed weapon you would see the gun above the body, a hand above the gun where the triger is, and a second hand out holding the other end of the gun behind the body.  If you have your hook out your second hand would hold the hook while the first hand would still be holding the gun.

This tread is not about flipping the tees body or feet. If you want to argue about flipping things I suggest you move to either https://github.com/teeworlds/teeworlds/issues/1399 or https://github.com/teeworlds/teeworlds/issues/1400 and we can talk about that there. please stop hijacking this thread.

**Comment by TheRealOrangus** [comment]
> This tread is not about flipping the tees body or feet
I know that, and I talk about flipping of hands when the aim cursor gets over the tee.

Well actually I like the idea, but it have to me smoother, like 1 hand is near the tee, and another is a bit farther, as you drawed, but it rotates that way, with a small shifting depending to aim direction, and no flipping of hands at all.

**Comment by android272** [comment]
No the hands would not flip. How ever hands are done now that is how the second hand would work.  There would just be two hands to make tees look better.  If you look at fan art and even art on the main site Tees have two hands. but when you play the game they do not.

**Comment by TheRealOrangus** [comment]
> If you look at fan art and even art on the main site Tees have two hands. **but when you play the game they do not.**

The second hand is just hidden until you use the hook :D

**Comment by android272** [comment]
On the home page.
![](https://www.teeworlds.com/images/splashtee6.png)

**Comment by TheRealOrangus** [comment]
As far as hands won't flip, as on images you(or someone else, dunno) drawed, it would be a nice graphical feature.

**Comment by android272** [comment]
I just tested a skin with a transparent body and the hand rotates around and does not flip.  The way the skin has the shadow you can not see it because it is behind the body. but if you were to see the hand it would look weird as it stands now.  I think that it would be best if there were a left and right hand in the sprites as well.  Otherwise the shadow would look as though it was being flipped even though it is only being rotated around the body.

**Comment by android272** [comment]
I just thought about that a bit more.  The left and right hands would not necessarily be the top and bottom hands as see above but the hand used when the Tee is looking left or right.  If you were to use the "left" hand for the back and the "right" hand for the front the shadows would be on opposite sides.  But instead when the Tee is looking left the "left" hand should be used for both the top and bottom hands and when the Tee is looking right the "right" hand would be used for both the top and bottom hands.  This way the shadow would be consistent between the two hands.

**Comment by TheRealOrangus** [comment]
That _is_ flipping, if I got what you said right.
Well, on my opinion this would be better with _"far"_ and _"near"_ hands, they'd just rotate(like it is now) and shift a bit, depends to aiming angle.

**Comment by android272** [comment]
![second hand](https://cloud.githubusercontent.com/assets/5912527/10680719/3137c9f0-78f0-11e5-86be-199d7f39270c.gif)
![second hand2](https://cloud.githubusercontent.com/assets/5912527/10680720/33666fb0-78f0-11e5-9980-aa8912ca980e.gif)

The problem with just rotating the hands is that the shadow that was on the bottom of the hand when the tee was looking right is now on the top of the hand when the tee looks left.  So I think that you were right when you said before that there needs to be two hands, a left and right hand.

**Comment by ObaniGemini** [comment]
Wouldn't it cause problems with custom gameskins ? I mean if a weapon is bigger than default and tha the hand is not well placed...

**Comment by TheRealOrangus** [comment]
> Wouldn't it cause problems with custom gameskins ? I mean if a weapon is bigger than default and tha the hand is not well placed...

There could be some config where are hands placed.

> *some images here..*
The problem with just rotating the hands is that the shadow that was on the bottom of the hand when the tee was looking right is now on the top of the hand when the tee looks left. So I think that you were right when you said before that there needs to be two hands, a left and right hand.

* Don't remember I was talking there should be *right & left*, because *You* said about *right & left*, and I was saying about *far & near*
* This is simple, just flip _image of the hand(not the hand itself!)_

**Comment by android272** [comment]
>Wouldn't it cause problems with custom gameskins ? I mean if a weapon is bigger than default and tha the hand is not well placed...

I guess this could look odd. But if this was the way it was then people would have to change their custom skins.  I don't know if it would look odd though.

>There could be some config where are hands placed.

I don't know how hands are placed now so I can't give input on this.

> * Don't remember I was talking there should be right & left, because You said about right & left, and I was saying about far & near

> * This is simple, just flip image of the hand(not the hand itself!)

Isn't this what you have been saying is wrong with flipping the skins all along?
